File: CS/RangeDate/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;


namespace DateRange
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            myGridView1.FilterPopupExcelData += MyGridView1_FilterPopupExcelData;
        }

        private void MyGridView1_FilterPopupExcelData(object sender, DevExpress.XtraGrid.Views.Grid.FilterPopupExcelDataEventArgs e) {
            if(e.Column.FieldName == "Range Date")
            e.AddFilter("Custom filter", String.Format("[{1}] = #{0}#", DateTime.Today.Date, e.Column.FieldName));
        }

        RepositoryItemDateEdit riDateEdit = new RepositoryItemDateEdit();
        private void Form1_Load(object sender, EventArgs e)
        {
            DataTable dataTable = new DataTable();
            dataTable.Columns.Add("Range Date", typeof(DateTime));
            dataTable.Columns.Add("Event", typeof(string));
            object[] row0 = new object[2] { DateTime.Today - new TimeSpan(1, 0, 0, 0), "Yesterday" };
            object[] row1 = new object[2] { DateTime.Today, "Today" };
            object[] row2 = new object[2] { DateTime.Today + new TimeSpan(1, 0, 0, 0), "Tomorrow" };
            dataTable.Rows.Add(row0);
            dataTable.Rows.Add(row1);
            dataTable.Rows.Add(row2);
            myGridControl1.DataSource = dataTable;
            myGridView1.Columns["Range Date"].ColumnEdit = riDateEdit;
        }
    }
}

File: VB/RangeDate/Form1.vb
Imports System
Imports System.Data
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository

Namespace DateRange

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            AddHandler myGridView1.FilterPopupExcelData, AddressOf MyGridView1_FilterPopupExcelData
        End Sub

        Private Sub MyGridView1_FilterPopupExcelData(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.FilterPopupExcelDataEventArgs)
            If Equals(e.Column.FieldName, "Range Date") Then e.AddFilter("Custom filter", String.Format("[{1}] = #{0}#", Date.Today.Date, e.Column.FieldName))
        End Sub

        Private riDateEdit As RepositoryItemDateEdit = New RepositoryItemDateEdit()

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim dataTable As DataTable = New DataTable()
            dataTable.Columns.Add("Range Date", GetType(Date))
            dataTable.Columns.Add("Event", GetType(String))
            Dim row0 As Object() = New Object(1) {Date.Today - New TimeSpan(1, 0, 0, 0), "Yesterday"}
            Dim row1 As Object() = New Object(1) {Date.Today, "Today"}
            Dim row2 As Object() = New Object(1) {Date.Today + New TimeSpan(1, 0, 0, 0), "Tomorrow"}
            dataTable.Rows.Add(row0)
            dataTable.Rows.Add(row1)
            dataTable.Rows.Add(row2)
            myGridControl1.DataSource = dataTable
            myGridView1.Columns("Range Date").ColumnEdit = riDateEdit
        End Sub
    End Class
End Namespace
